src/Couchbase.Lite.Shared/Query/XQuery.cs

```diff
@@ -207,13 +207,12 @@ private unsafe void Check()
             var columnCnt = Native.c4query_columnCount(query);
             for (int i = 0; i < columnCnt; i++) {
                 var titleStr = Native.c4query_columnTitle(query, (uint)i).CreateString();
-                if(titleStr=="*" || titleStr.StartsWith("* #")) {
-                    if(selectListCnt == columnCnt)
-                        titleStr = selectResultList.ElementAtOrDefault(i)?.ColumnName;
-                }
 
-                if (String.IsNullOrEmpty(titleStr))
-                    titleStr = FromImpl.ColumnName;
+                if (titleStr.StartsWith("*")) {
+                    titleStr = selectResultList.ElementAtOrDefault(i)?.ColumnName;
+                    if (String.IsNullOrEmpty(titleStr))
+                        titleStr = FromImpl.ColumnName;
```

Comment:
Double check that this is needed....I don't think it is

Reply:
Actually it was hit several times...

Reply:
Interesting.  Was there any times that line 211 was true but line 213 was false?  Could we remove the "column name" completely?

Reply:
Yes, few times that line 211 was true but line 213 was false. Cannot remove "column name" completely.

Reply:
I'm not getting why we need to have both.....It seems overly complicated.  Can you give examples of each case?